contrib/openapi/spec.yml

```diff
@@ -1,453 +1,282 @@
-            - fsharp
-            - gas
-            - gherkin
-            - glsl
-            - gnuplot
-            - go
-            - groovy
-            - haskell
-            - html
-            - html+php
-            - ini
-            - io
-            - java
-            - javascript
-            - lighttpd
-            - llvm
-            - lua
-            - makefile
-            - matlab
-            - minid
-            - modelica
-            - modula2
-            - moonscript
-            - mupad
-            - nasm
-            - nemerle
-            - newlisp
-            - nginx
-            - nimrod
-            - objectivec
-            - ocaml
-            - octave
-            - opa
-            - perl
-            - postscript
-            - powershell
-            - prolog
-            - pycon
-            - python
-            - python3
-            - rbcon
-            - rebol
-            - restructuredtext
-            - ruby
-            - scala
-            - scheme
-            - scilab
-            - smalltalk
-            - smarty
-            - splus
-            - sql
-            - systemverilog
-            - tcl
-            - tex
-            - text
-            - vala
-            - vb.net
-            - verilog
-            - vhdl
-            - vim
-            - xml
-            - yaml
-          required: false
-          description: Only return snippets of the given syntax.
-        - name: author
-          in: query
-          type: integer
-          minimum: 0
-          required: false
-          description: Only return snippets by the given author.
-        - name: tag
-          in: query
-          type: string
-          pattern: '(\w+)(,\w+)*'
-          required: false
-          description: Only return snippets that have the given tag(s) assigned.
         - name: marker
           in: query
           type: integer
-          minimum: -1
-          default: -1
+          minimum: 1
           format: int64
-          required: false
           description: |
-            The id of the last seen snippet. Used for pagination. Will return
-            `limit` number of snippets after this one.
+            Last seen snippet that is used in [pagination](#section/Pagination)
+            to return `limit` number of snippets after this one.
         - name: limit
           in: query
           type: integer
           format: int32
           minimum: 1
-          maximum: 1000
-          default: 20
-          required: false
-          description: |
-            The maximum number of snippets to be returned by this query.
+          maximum: 100
+          default: 10
+          description: A limit on the number of snippets to be returned.
       responses:
         200:
-          description: An array of code snippets
+          description: OK
           schema:
             type: array
             items:
               $ref: '#/definitions/Snippet'
-        400:
```

Comment:
You are right, my mistake. I'll return both 400 and 404 (since we return 404 if `marker` points to a non-existing snippet).